lib_eio_linux/eio_linux.mli

```diff
@@ -248,4 +248,29 @@ module Low_level : sig
   (** [getaddrinfo host] returns a list of IP addresses for [host]. [host] is either a domain name or
       an ipaddress. *)
 
+  module Process : sig
+    type t
+    (** A subprocess *)
+
+    val spawn :
+      ?env:Spawn.Env.t ->
```

Comment:
Process management is a pretty core function of eio.  Given that `spawn` is just a single set of ml/mli/c files, perhaps we should just vendor it in here in Eio_linux and give local assignments to the Env and Working_dir modules, to keep Eio more standalone.

Reply:
Seems reasonable to me, I think (though I haven't tried it yet) that would also mean switching over to `clone3` would be fairly straightforward too.

Reply:
I'm experimenting with vendoring it over in https://github.com/patricoferris/eio/commit/c8a28ebca940d5980fa39f7e47cf33f04796f0bc for the moment